### PR TITLE
test(pectra): strengthen crypto-primitive test assertions

### PR DESCRIPTION
**Pull request by CWagamanEure**
Strengthens assertions in three Pectra crypto-primitive test files that had full coverage but verified little: a tautological amount-leaf check, a missing gwei-alignment revert, an unexercised pubkey leaf, and a hollow idempotence test. Also anchors the deposit domain to the published mainnet constant. Test-only (no src/ changes); all tests pass under via_ir.

Resubmitted on the main repo; supersedes the earlier fork-based PR #616.

## Summary by CodeRabbit

* **Tests**
  * Strengthened BLS12_381 SSZ validation test coverage with new deposit domain and signing-root tests
  * Added edge case tests for non-gwei-aligned amounts and non-zero pubkeys
  * Enhanced test documentation and clarity in pubkey lookup validation

### CODE REVIEW

**Comment by CWagamanEure** [comment]
@coderabbitai review

File: contracts/test/libraries/BLS12_381_SSZ.t.sol [hunk 1]
```diff
@@ -70,17 +83,20 @@ contract BLS12_381_SSZTest is Test {
...
         bytes32 wc = keccak256("withdrawal_credentials");
         bytes32 domain = keccak256("domain");
         uint256 amount = 32 ether;
-        uint256 amountGwei = amount / 1 gwei;
 
         // Spec-correct computation inline
         bytes32 pubkeyRoot = sha256(abi.encodePacked(pubkey, bytes16(0)));
-        bytes32 amountChunk = bytes32(LibUint256.toLittleEndian64(amountGwei));
+        // 32 ETH = 32e9 gwei = 0x0040597307000000 little-endian (independent of toLittleEndian64)
+        bytes32 amountChunk = 0x0040597307000000000000000000000000000000000000000000000000000000;
```
Comment (juliaaschmidt) [inline contracts/test/libraries/BLS12_381_SSZ.t.sol]: Converting to bytes everywhere makes this less humanly readable, I'd prefer to keep it the old way.

File: contracts/test/libraries/BLS12_381_SSZ.t.sol [hunk 2]
```diff
@@ -58,6 +58,20 @@ contract BLS12_381_SSZTest is Test {
         assertEq(actualDomain, expectedDomain, "computeDepositDomain holesky does not match spec");
     }
 
+    // Anchors computeDepositDomain to an external published value rather than to a
+    // re-derivation in this test. For mainnet (genesis fork version 0x00000000 and a
+    // zero genesis_validators_root), the ForkData root is the canonical SSZ zero hash
+    // z1 = sha256(64 zero bytes) = f5a5fd42...2759fb4b, so the deposit domain is
+    // DOMAIN_DEPOSIT (0x03000000) followed by z1's first 28 bytes.
+    function test_computeDepositDomain_mainnetVector() public {
```
Comment (juliaaschmidt) [inline contracts/test/libraries/BLS12_381_SSZ.t.sol]: this looks like a duplicate test of the existing `test_computeDepositDomain_matchesSpec`

File: contracts/test/libraries/BLS12_381_SSZ.t.sol [hunk 3]
```diff
@@ -111,4 +125,45 @@ contract BLS12_381_SSZTest is Test {
         bytes32 actualSigningRoot = harness.computeSigningRoot(pubkey, amount, wc, domain);
         assertEq(actualSigningRoot, expectedSigningRoot, "signing root does not match spec for 1 ETH deposit");
     }
+
+    // Verifies depositMessageSigningRoot reverts with InvalidDepositAmount when the
+    // deposit amount is not a whole number of gwei (BLS12_381.sol:584). Uses
+    // 1 gwei + 1 wei; the aligned-amount happy path is covered by the tests above.
+    function test_computeSigningRoot_revertsOnNonGweiAlignedAmount() public {
```
Comment (juliaaschmidt) [inline contracts/test/libraries/BLS12_381_SSZ.t.sol:115]: Nice test! 👍

File: contracts/test/libraries/BLS12_381_SSZ.t.sol [hunk 4]
```diff
@@ -111,4 +125,45 @@ contract BLS12_381_SSZTest is Test {
         bytes32 actualSigningRoot = harness.computeSigningRoot(pubkey, amount, wc, domain);
         assertEq(actualSigningRoot, expectedSigningRoot, "signing root does not match spec for 1 ETH deposit");
     }
+
+    // Verifies depositMessageSigningRoot reverts with InvalidDepositAmount when the
```
Comment (juliaaschmidt) [inline contracts/test/libraries/BLS12_381_SSZ.t.sol]: Do we need this description? The test name should be descriptive enough to explain what is tested. I don't think the majority of tests has this comments section?

File: contracts/test/libraries/BLS12_381_SSZ.t.sol [hunk 5]
```diff
@@ -111,4 +125,45 @@ contract BLS12_381_SSZTest is Test {
...
+        bytes memory pubkey = new bytes(48);
+        bytes32 wc = bytes32(uint256(1));
+        bytes32 domain = bytes32(uint256(0xdeadbeef));
+        uint256 amount = 1 gwei + 1;
+
+        vm.expectRevert(BLS12_381.InvalidDepositAmount.selector);
+        harness.computeSigningRoot(pubkey, amount, wc, domain);
+    }
+
+    // Verifies the signing root for a fully non-zero 48-byte pubkey (bytes 0x01..0x30).
```
Comment (juliaaschmidt) [inline contracts/test/libraries/BLS12_381_SSZ.t.sol]: same here

**Review by juliaaschmidt — APPROVED** [review]
